### PR TITLE
Customizer now uses React 16 context.

### PR DESCRIPTION
**Pull request by dzearing**
Addresses #5691 and #5413.

We were observing some peculiarities with theme changing. Addressed multiple issues here:

1. We now use react16 context rather than the deprecated context for customizer. We *think* there may have been issues in the old context model wrt pure components not propagating context changes to children, but now I'm not completely sure. Nevertheless, less deprecated api usage is good.

2. Check shouldComponentupdate was too restrictive and didn't consider mutations to theme or classname. Adjusting logic fixed some theming issues we saw.

3. When `loadTheme` is called to apply a global theme, we were not observing mutations to this, so Customizers wouldn't re-evaluate and re-render. We now do observe global theme changes and rerender.

4. Minor tweak to webpack config.

All customizer tests still pass, and DetailsList now respects

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Check/Check.base.tsx [hunk 1]
```diff
@@ -14,7 +14,11 @@ export class CheckBase extends BaseComponent<ICheckProps, {}> {
   };
 
   public shouldComponentUpdate(newProps: ICheckProps): boolean {
-    return this.props.checked !== newProps.checked;
+    return (
+      this.props.className !== newProps.className ||
+      this.props.checked !== newProps.checked ||
+      this.props.theme !== newProps.theme
+    );
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/Check/Check.base.tsx]: What if styles prop changes? I feel like having shouldComponentUpdate for such a simple component is code smell and we're checking most of the props now. I'm curious why it exists at all

File: packages/utilities/src/customizable.tsx [hunk 2]
```diff
@@ -39,11 +33,14 @@ export function customizable(
       }
 
       public render(): JSX.Element {
-        const defaultProps = Customizations.getSettings(fields, scope, this.context.customizations);
-
         return (
-          // tslint:disable-next-line:no-any
-          <ComposedComponent {...defaultProps} {...this.props as any} />
+          <CustomizerContext.Consumer>
```
Comment (JasonGore) [inline packages/utilities/src/customizable.tsx]: 👍 I can steal this for createComponent 😄

File: apps/todo-app/package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "office-ui-fabric-react": ">=6.40.0 <7.0.0",
     "react": ">=16.3.2-0 <17.0.0",
     "react-dom": ">=16.3.2-0 <17.0.0",
-    "typescript": "2.8.4",
+    "typescript": "3.0.0-rc",
```
Comment (cliffkoh) [inline apps/todo-app/package.json]: ?

**Comment by dzearing** [comment]
Looks like I may be blocked on this getting published in `enzyme-adapter-react-16`.

https://github.com/airbnb/enzyme/issues/1620

**Comment by cliffkoh** [comment]
I think you have created a separate PR to get a smaller subset of this PR in, while the rest of the changes are blocked on enzyme fixing a bug...

I will close this then until you can actually make progress on it.

**Pull request merged**